torch/_prims/__init__.py

```diff
@@ -1814,19 +1814,17 @@ def _as_strided_scatter_meta(
     utils.validate_shape(size)
     utils.validate_strides(stride)
 
-    input_length = utils.compute_required_storage_length(
-        input.size(), input.stride(), input.storage_offset()
-    )
-    required_view_length = utils.compute_required_storage_length(
+    input_size = input.untyped_storage().size()
```

Comment:
how would this work for fake tensors?

Reply:
I tested with original repro and observed that fake tensors can return the corrected size with input.untyped_storage().size(). 

I also test below:

```
>>> with FakeTensorMode():
...        x = torch.rand((4, 1))
...
>>> x
FakeTensor(FakeTensor(..., device='meta', size=(4, 1)), cpu)
>>> x.untyped_storage()
...
[torch.storage.UntypedStorage(device=meta) of size 16]

>>> x.untyped_storage().size()
16
>>> x[1].untyped_storage().size()
16
```

I can add a unit test in fake tensor. Do you see any concern using this to determine the base tensor's size of a view? Any suggestion on proper way to check the size? Appreciate your help!